oci/layout/oci_dest.go

```diff
@@ -299,6 +304,102 @@ func (d *ociImageDestination) CommitWithOptions(ctx context.Context, options pri
 	return os.WriteFile(d.ref.indexPath(), indexJSON, 0644)
 }
 
+// tryReflinkLocalFile attempts to reflink the specified file and digest it.
+// If relinking does not work, reset to doing a verbatim copy of the file.
+func tryReflinkLocalFile(dest *ociImageDestination, file string) (private.UploadedBlob, bool, error) {
+	fInfo, err := os.Stat(file)
+	if err != nil {
+		return private.UploadedBlob{}, false, err
+	}
+
+	blobFile, err := os.CreateTemp(dest.ref.dir, "oci-put-blob")
+	if err != nil {
+		return private.UploadedBlob{}, false, err
+	}
+	blobName := blobFile.Name()
+
+	copyRange := false
+	copyClone := true
+	err = reflinkCopy.CopyRegularToFile(file, blobFile, fInfo, &copyRange, &copyClone)
+	if err != nil {
+		return private.UploadedBlob{}, false, err
+	}
+
+	_, err = blobFile.Seek(0, 0)
+	if err != nil {
+		return private.UploadedBlob{}, false, err
+	}
+
+	blobFile, err = os.Open(blobName)
+	if err != nil {
+		return private.UploadedBlob{}, false, err
+	}
+	blobDigest, err := digest.FromReader(blobFile)
+	if err != nil {
+		blobFile.Close()
+		return private.UploadedBlob{}, false, err
+	}
+	blobPath, err := dest.ref.blobPath(blobDigest, dest.sharedBlobDir)
```

Comment:
I’d prefer to share the implementation with `PutBlobWithOptions` from now on, so that the `Sync` and `Chmod` behavior, and any future additions to `PutBlobWithOptions`, are kept in sync.